kernel/thread.c

```diff
@@ -853,6 +853,27 @@ int z_stack_space_get(const uint8_t *stack_start, size_t size, size_t *unused_pt
 		size -= 4;
 	}
 
+#ifdef CONFIG_STACK_ALIGN_DOUBLE_WORD
+	const uint64_t *checked_stack_word = (const uint64_t *)checked_stack;
```

Comment:
The stack on a 64-bit platform has at least 8-byte memory alignment. Adding 4 bytes to `checked_stack` in the CONFIG_STACK_SENTINEL case in the preceding code block will change `checked_stack` to be 4-byte aligned. This has the potential to generate an alignment exception when reading a 64-bit value from a 32-bit aligned memory address. Whether such an exception would occur would be both architecture and compiler dependent.